lib/src/themes/expression/parser/categorical_expression_parser.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:vector_tile_renderer/src/themes/expression/property_expression.dart';
+
+import '../expression.dart';
+import 'expression_parser.dart';
+
+class CategoricalExpressionParser extends ExpressionComponentParser {
+  CategoricalExpressionParser(ExpressionParser parser)
+      : super(parser, 'categorical');
+
+  @override
+  Expression? parse(List<dynamic> json) {
+    // [0] expression type
+    // [1] property name
+    // [2] default value
+    // [3:] pair of stops
+    if (json.length < 3 || json.length.isEven) {
+      throw Exception(
```

Comment:
this library is intended to fail gracefully, we don't want to throw here

instead, return null and the library will log that the expression isn't supported